### ISSUE TITLE
bootstrap_kmaster task 1 is not able to pull

### ISSUE THREAD

**Issue by keobox** [justmeandopensource/kubernetes#126]
**Describe the bug**
`vagrant up` does not bring up a fresh cluster.

**How To Reproduce**
`vagrant ssh kmaster`
`kubeadm config images pull`

**Expected behavior**
Image pulled.

**Screenshots (if any)**
root@kmaster:/home/vagrant# kubeadm config images pull
I0119 11:01:22.159857    5007 version.go:255] remote version is much newer: v1.26.1; falling back to: stable-1.24
failed to pull image "k8s.gcr.io/kube-apiserver:v1.24.9": output: time="2023-01-19T11:01:22Z" level=fatal msg="validate service connection: CRI v1 image API is not implemented for endpoint \"unix:///var/run/containerd/containerd.sock\": rpc error: code = Unimplemented desc = unknown service runtime.v1.ImageService"
, error: exit status 1

**Environment (please complete the following information):**
MacOS Ventura and Vagrant 7.0.6

**Additional context**
root@kmaster:/home/vagrant# kubeadm config images pull --v=5
I0119 11:06:52.015445    5239 initconfiguration.go:117] detected and using CRI socket: unix:///var/run/containerd/containerd.sock
I0119 11:06:52.015652    5239 interface.go:432] Looking for default routes with IPv4 addresses
I0119 11:06:52.015690    5239 interface.go:437] Default route transits interface "eth0"
I0119 11:06:52.015778    5239 interface.go:209] Interface eth0 is up
I0119 11:06:52.016164    5239 interface.go:257] Interface "eth0" has 1 addresses :[10.0.2.15/24].
I0119 11:06:52.016647    5239 interface.go:224] Checking addr  10.0.2.15/24.
I0119 11:06:52.016696    5239 interface.go:231] IP found 10.0.2.15
I0119 11:06:52.016736    5239 interface.go:263] Found valid IPv4 address 10.0.2.15 for interface "eth0".
I0119 11:06:52.016761    5239 interface.go:443] Found active IP 10.0.2.15 
I0119 11:06:52.016824    5239 kubelet.go:214] the value of KubeletConfiguration.cgroupDriver is empty; setting it to "systemd"
I0119 11:06:52.022107    5239 version.go:186] fetching Kubernetes version from URL: https://dl.k8s.io/release/stable-1.txt
I0119 11:06:52.581213    5239 version.go:255] remote version is much newer: v1.26.1; falling back to: stable-1.24
I0119 11:06:52.581923    5239 version.go:186] fetching Kubernetes version from URL: https://dl.k8s.io/release/stable-1.24.txt
exit status 1
output: time="2023-01-19T11:06:53Z" level=fatal msg="validate service connection: CRI v1 image API is not implemented for endpoint \"unix:///var/run/containerd/containerd.sock\": rpc error: code = Unimplemented desc = unknown service runtime.v1.ImageService"
, error
k8s.io/kubernetes/cmd/kubeadm/app/util/runtime.(*CRIRuntime).PullImage
	cmd/kubeadm/app/util/runtime/runtime.go:121
k8s.io/kubernetes/cmd/kubeadm/app/cmd.PullControlPlaneImages
	cmd/kubeadm/app/cmd/config.go:340
k8s.io/kubernetes/cmd/kubeadm/app/cmd.newCmdConfigImagesPull.func1
	cmd/kubeadm/app/cmd/config.go:312
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).execute
	vendor/github.com/spf13/cobra/command.go:856
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).ExecuteC
	vendor/github.com/spf13/cobra/command.go:974
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).Execute
	vendor/github.com/spf13/cobra/command.go:902
k8s.io/kubernetes/cmd/kubeadm/app.Run
	cmd/kubeadm/app/kubeadm.go:50
main.main
	cmd/kubeadm/kubeadm.go:25
runtime.main
	/usr/local/go/src/runtime/proc.go:250
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1571
failed to pull image "k8s.gcr.io/kube-apiserver:v1.24.9"
k8s.io/kubernetes/cmd/kubeadm/app/cmd.PullControlPlaneImages
	cmd/kubeadm/app/cmd/config.go:341
k8s.io/kubernetes/cmd/kubeadm/app/cmd.newCmdConfigImagesPull.func1
	cmd/kubeadm/app/cmd/config.go:312
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).execute
	vendor/github.com/spf13/cobra/command.go:856
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).ExecuteC
	vendor/github.com/spf13/cobra/command.go:974
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).Execute
	vendor/github.com/spf13/cobra/command.go:902
k8s.io/kubernetes/cmd/kubeadm/app.Run
	cmd/kubeadm/app/kubeadm.go:50
main.main
	cmd/kubeadm/kubeadm.go:25
runtime.main
	/usr/local/go/src/runtime/proc.go:250
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1571

**Comment by keobox** [comment]
There's a sort of connection issue, it seems
validate service connection: CRI v1 image API is not implemented for endpoint "unix:///var/run/containerd/containerd.sock

**Comment by alugbinPSQR** [comment]
https://serverfault.com/questions/1118051/failed-to-run-kubelet-validate-service-connection-cri-v1-runtime-api-is-not-im

The approved answer explained the issue.
TLDR: The problem is that the apt for `containerd` is kind of outdated, hence the versions doesn't match

**Comment by keobox** [comment]
Yep! I've found this myself this morning, seems something introduced in containerd version 1.5.9, because before it was working. I did the manual containerd 1.6.12 installation and it worked.

**Comment by keobox** [comment]
This issue can be close, since the link above works.
